rpcs3/rpcs3qt/gs_frame.cpp

```diff
@@ -299,6 +302,96 @@ void gs_frame::flip(draw_context_t, bool /*skip_frame*/)
+{
+	std::thread(
+	    [ss_width, ss_height, vulkan_buf](std::vector<u8> sshot_data) {
+		    std::string screen_path = fmt::format("%s/Screenshots/", fs::get_config_dir());
+
+		    if (!fs::is_dir(screen_path))
+		    {
+			    if (!fs::create_dir(screen_path))
+			    {
+				    LOG_ERROR(GENERAL, "Failed to create screenshot path: %s", screen_path);
+				    return;
+			    }
+		    }
+
+		    u32 file_index;
+		    for (file_index = 0; file_index < 100000000; file_index++)
+		    {
+			    std::string filename = fmt::format("%sScreenshot-%08d.png", screen_path, file_index);
+			    if (fs::is_file(filename))
+				    continue;
+
+			    fs::file ss_file(filename, fs::open_mode::create + fs::open_mode::write);
+			    if (!ss_file)
+			    {
+				    LOG_ERROR(GENERAL, "[Screenshot] Failed to save screenshot to %s", filename);
+				    return;
+			    }
+
+			    std::vector<u8> ss_data_alpha(sshot_data.size());
+			    const u32* ss_ptr = (const u32*)sshot_data.data();
+			    u32* alpha_ptr    = (u32*)ss_data_alpha.data();
+
+			    if (vulkan_buf)
+			    {
+				    for (size_t index = 0; index < sshot_data.size() / sizeof(u32); index++)
+				    {
+					    alpha_ptr[index] = ((ss_ptr[index] & 0xFF) << 16) | (ss_ptr[index] & 0xFF00) | ((ss_ptr[index] & 0xFF0000) >> 16) | 0xFF000000;
+				    }
+			    }
+			    else
+			    {
+				    for (size_t index = 0; index < sshot_data.size() / sizeof(u32); index++)
+				    {
+					    alpha_ptr[index] = ss_ptr[index] | 0xFF;
+				    }
+			    }
+
+			    std::vector<u8> encoded_png;
+
+			    png_structp write_ptr = png_create_write_struct(PNG_LIBPNG_VER_STRING, nullptr, nullptr, nullptr);
+			    png_infop info_ptr    = png_create_info_struct(write_ptr);
+			    png_set_IHDR(write_ptr, info_ptr, ss_width, ss_height, 8, PNG_COLOR_TYPE_RGBA, PNG_INTERLACE_NONE, PNG_COMPRESSION_TYPE_DEFAULT, PNG_FILTER_TYPE_DEFAULT);
+
+			    std::vector<u8*> rows(ss_height);
+			    for (size_t y = 0; y < ss_height; y++)
+				    rows[y] = (u8*)ss_data_alpha.data() + y * ss_width * 4;
+
+			    png_set_rows(write_ptr, info_ptr, &rows[0]);
+			    png_set_write_fn(write_ptr, &encoded_png,
+			        [](png_structp png_ptr, png_bytep data, png_size_t length) {
+				        std::vector<u8>* p = (std::vector<u8>*)png_get_io_ptr(png_ptr);
+				        p->insert(p->end(), data, data + length);
+			        },
+			        nullptr);
+
+			    if (vulkan_buf)
+				    png_write_png(write_ptr, info_ptr, PNG_TRANSFORM_IDENTITY, nullptr);
+			    else
+				    png_write_png(write_ptr, info_ptr, PNG_TRANSFORM_SWAP_ALPHA, nullptr);
+
+			    png_free_data(write_ptr, info_ptr, PNG_FREE_ALL, -1);
+			    png_destroy_write_struct(&write_ptr, nullptr);
+
+			    ss_file.write(encoded_png.data(), encoded_png.size());
+
+			    LOG_SUCCESS(GENERAL, "[Screenshot] Successfully saved screenshot to %s", filename);
+			    return;
+		    }
+
+		    if (file_index == 100000000)
```

Comment:
Does this check need to be here? I don't see how you can get to this codepath without reaching the filecount limit.